hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -834,6 +835,12 @@ public DirectoryListing getListing(String src, byte[] startAfter,
     return new DirectoryListing(combinedData, remainingEntries);
   }
 
+  @Override
+  public BatchedDirectoryListing getBatchedListing(String[] srcs,
+      byte[] startAfter, boolean needLocation) throws IOException {
+    throw new UnsupportedOperationException("Not implemented");
```

Comment:
let's file a jira to support this API in RBF.

Reply:
Done. Filed https://issues.apache.org/jira/browse/HDFS-15029